### PR TITLE
[fix] Fix code block content and query copying functionality

### PR DESCRIPTION
**Pull request by arsengit**
- Fix CodeBlock copying functionality
- Tuned `IllustrationBlock` styles
- Added `useCopy` hooks
- Added `onCopyToClipBoard` helper function

### CODE REVIEW

**Comment by arsengit** [comment]
Resolves #2079

File: aim/web/ui/src/hooks/useCopy/useCopy.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import { ICopyToClipBoardProps } from 'types/components/CopyToClipBoard/CopyToClipBoard';
+
+import { onCopyToClipBoard } from 'utils/onCopyToClipBoard';
+
+export function useCopy({
+  contentRef,
```
Comment (KaroMourad) [inline aim/web/ui/src/hooks/useCopy/useCopy.ts:8]: let's use a simple argument not an object, as it's a single argument

File: aim/web/ui/src/hooks/useCopy/useCopy.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}: {
+  contentRef: ICopyToClipBoardProps['contentRef'];
+}) {
+  const [copied, setCopied] = React.useState(false);
+  function onCopy() {
```
Comment (KaroMourad) [inline aim/web/ui/src/hooks/useCopy/useCopy.ts:13]: let's memoize the function to improve performance

File: aim/web/ui/src/utils/onCopyToClipBoard.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      const textArea: HTMLTextAreaElement = document.createElement('textarea');
+      textArea.value = text;
+
+      // make the textarea out of viewport
+      textArea.style.position = 'fixed';
+      textArea.style.left = '-400vw';
+      textArea.style.top = '-400vh';
+      document.body.appendChild(textArea);
+      textArea.focus();
+      textArea.select();
```
Comment (KaroMourad) [inline aim/web/ui/src/utils/onCopyToClipBoard.ts:15]: let's remove `textarea` element  from the document after copying the text

**Review by KaroMourad — APPROVED** [review]
great work

**Pull request merged**